docs/reference/buildx_bake.md

```diff
@@ -211,18 +320,83 @@ Complete list of valid target fields:
 `args`, `cache-from`, `cache-to`, `context`, `dockerfile`, `inherits`, `labels`,
 `no-cache`, `output`, `platform`, `pull`, `secrets`, `ssh`, `tags`, `target`
 
+### Global scope attributes
+
+You can define global scope attributes in HCL/JSON and use them for code reuse and
+setting values for variables. This means you can do a "data-only" HCL file with the
+values you want to set/override and use it in the list of regular output files:
+
+```hcl
+# docker-bake.hcl
+variable "FOO" {
+    default = "abc"
+}
+
+target "app" {
+    args = {
+        v1 = "pre-${FOO}"
+    }
+}
+```
+
+You can use this file directly:
+
+```console
+$ docker buildx bake --print app
+{
+  "target": {
+    "app": {
+      "context": ".",
+      "dockerfile": "Dockerfile",
+      "args": {
+        "v1": "pre-abc"
+      }
+    }
+  }
+}
+```
+
+Or create an override configuration file:
+
+```hcl
+# env.hcl
+WHOAMI="myuser"
+FOO="def-${WHOAMI}"
+```
+
+And invoke bake together with both of the files:
+
+```console
+$ docker buildx bake -f docker-bake.hcl -f env.hcl --print app
+{
+  "target": {
+    "app": {
+      "context": ".",
+      "dockerfile": "Dockerfile",
+      "args": {
+        "v1": "pre-def-myuser"
+      }
+    }
+  }
+}
+```
+
 ### HCL variables and functions
 
 Similar to how Terraform provides a way to [define variables](https://www.terraform.io/docs/configuration/variables.html#declaring-an-input-variable),
 the HCL file format also supports variable block definitions. These can be used
 to define variables with values provided by the current environment, or a default
 value when unset.
 
+A [set of generally useful functions](https://github.com/docker/buildx/blob/master/bake/hclparser/stdlib.go)
+provided by [go-cty](https://github.com/zclconf/go-cty/tree/main/cty/function/stdlib)
+are available for use in HCL files. In addition, [user defined functions](https://github.com/hashicorp/hcl/tree/main/ext/userfunc)
+are also supported.
 
-Example of using interpolation to tag an image with the git sha:
+#### Using interpolation to tag an image with the git sha
```

Comment:
Not a show-stopper (would be ok for a follow-up), but it'd be nice to have a short introduction for each example. Admitted, with a descriptive "heading" it's a bit of "boiler plating", but it looks nice if there's a short introduction instead of a code-block directly after a heading, and it can lead in to what the example is illustrating.

Things along the lines of;

> The following example shows a docker-bake HCL file that specifies a `TAG` variable to allow customizing the build:

Reply:
Done